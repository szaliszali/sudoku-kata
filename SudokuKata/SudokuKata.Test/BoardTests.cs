using System;
using System.Collections.Generic;
using ApprovalTests;
using NUnit.Framework;

namespace SudokuKata.Test;

[TestFixture]
internal class BoardTests
{
    [Test]
    public void EmptyBoardToString() => Approvals.Verify(new CharArrayBoard());

    [Test]
    public void EmptyBoardCode() => Approvals.Verify(new CharArrayBoard().Code);

    [Test]
    public void RandomBoard()
    {
        var rng = new Random(1);
        var stateStack = new Stack<int[]>();
        var sut = new RandomBoard(rng, stateStack);
        Approvals.Verify((sut, sut.Code));
    }
}
