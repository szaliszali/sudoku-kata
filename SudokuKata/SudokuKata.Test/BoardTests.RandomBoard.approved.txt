(+---+---+---+
|586|479|213|
|431|562|879|
|279|138|564|
+---+---+---+
|695|347|182|
|748|215|396|
|123|896|457|
+---+---+---+
|364|781|925|
|857|923|641|
|912|654|738|
+---+---+---+, 586479213431562879279138564695347182748215396123896457364781925857923641912654738)